modeling/gbert_classifier.py

```diff
@@ -224,52 +260,112 @@ def make_model(label):
+    # > For example, if a dataset contains 100 positive and 300 negative examples of a single class, 
+    # > then pos_weight for the class should be equal to 300/100=3 . 
+    # > The loss would act as if the dataset contains 3*100=300 positive examples.
+    loss_fn = nn.BCEWithLogitsLoss(pos_weight=torch.FloatTensor([positive_class_weight])).to(device)
 
     history = defaultdict(list)
-    best_f1 = 0
+    best_fbeta = 0
     t = datetime.now().strftime("%y%m%d_%H%M")
     for epoch in range(EPOCHS):
         logger.info(f'Epoch {epoch + 1}/{EPOCHS}')
         logger.info('-' * 10)
-        train_f1, train_loss = train_epoch(
+        train_fbeta, train_metrics, train_params, train_loss = train_epoch(
             model,
             train_data_loader,
             loss_fn,
             optimizer,
             device,
             scheduler,
-            len(df_train)
         )
-        logger.info(f'Train loss {train_loss} F1 {train_f1}')
-        val_f1, val_loss = eval_model(
+        logger.info(f'Train loss {train_loss} Fbeta {train_fbeta}')
+        val_fbeta, val_metrics, val_params,  val_loss = eval_model(
             model,
             val_data_loader,
             loss_fn,
             device,
-            len(df_val)
         )
-        logger.info(f'Val   loss {val_loss} F1 {val_f1}')
+        logger.info(f'Val   loss {val_loss} Fbeta {val_fbeta}')
         print()
-        history['train_f1'].append(train_f1)
+        history['train_fbeta'].append(train_fbeta)
         history['train_loss'].append(train_loss)
-        history['val_f1'].append(val_f1)
+        history['val_fbeta'].append(val_fbeta)
         history['val_loss'].append(val_loss)
-        if val_f1 > best_f1:
-            file_name = f"./models/model_gbert_pool_{label}_{t}.bin"
+        if val_fbeta > best_fbeta:
+            file_name = f"./models/model_gbertbase_{label}_{t}.bin"
             torch.save(model.state_dict(), file_name)
-            best_f1 = val_f1
+            for k,v in val_metrics.items():
+                mlflow_logger.add_metric(k, v)
+            for k,v in val_params.items():
+                mlflow_logger.add_param(k, v)
+            for k,v in train_metrics.items():
+                mlflow_logger.add_metric(k, v)
+            for k,v in train_params.items():
+                mlflow_logger.add_param(k, v)
+            best_fbeta = val_fbeta
+
+    #############################################
+    #MLflow logging
+
+    constant_params = {
+                    'epochs': EPOCHS,
+                    'batch_size': BATCH_SIZE,
+                    'max_len': MAX_LEN,
+                }
+    mlflow_logger.add_tag("cycle4", True)
+    mlflow_logger.add_param("normalization", 'norm')
+    mlflow_logger.add_param("vectorizer", 'deepset/gbert-base')
+    mlflow_logger.add_param("model", "deepset/gbert-base")
+    # I'm re-using the grid_search_params field in order to no open too many mlflow columns
```

Comment:
typo: no -> not

Reply:
Typo resolved in b7d6ab255560e5719b55d6496850204160730750